src/http.ts

```diff
@@ -434,7 +441,7 @@ export async function getAnalysis(options: {
   // ?linters=false is still a truthy query value, if(includeLint === false) we have to avoid sending the value altogether
   const params = { severity, linters: includeLint || undefined };
 
-  const headers = { 'Session-Token': sessionToken, source };
+  const headers = { ...requestOptions?.headers, 'Session-Token': sessionToken, source };
```

Comment:
question: maybe worth using white list here? for all the headers we would like to send..? (thinking about future confusions of passing unnecessary data as part of the request options...

Reply:
@orkamara I am not sure what do you mean. As we are in TypeScript, current type is supporting `headers` key only. So all you can do is to add headers. If you would like to add something else, you would need to extend the type first.